### ISSUE TITLE
Annotations-advanced: annotationToFields traverse function skipping options with default value false or 0

### ISSUE THREAD

**Issue by robocop98** [highcharts/highcharts#16384]
#### Expected behaviour
All relevant options defined in `NavigationBindings.annotationsEditable` should be provided as fields by `NavigationBindings.prototype.annotationToFields`.

#### Actual behaviour
Options with boolean default value false are skipped.
Same would occur for number with default value of 0.

Culprit is the check for `&& option` in:
```
                function traverse(option, key, parentEditables, parent) {
                    var nextParent;
                    if (parentEditables &&
                        option &&
```
checking instead for `&& option != null `would allow boolean with default false and number with default 0.

Also the following call at end of `NavigationBindings.prototype.annotationToFields` has too many parameters. 
I think that either `visualOptions[key]` needs to be replaced with true, or true needs to be removed.
`traverse(typeOption, typeKey, nestedEditables, visualOptions[key], true);`

#### Live demo with steps to reproduce
n/a

#### Product version
Highstocks 9.2.2

#### Affected browser(s)
all

**Comment by raf18seb** [comment]
Hi @robocop98, thanks for reporting the issue!

**Steps to reproduce:**
As listed above, there are 2 bugs (the second bug is rather about useless parameter and code cleanliness, it isn't breaking anything):
- take this [Stock Tools gui](https://jsfiddle.net/BlackLabel/yeg9k1sj/) demo with a `navigation.bindings.rectangleAnnotation` annotation defined with a `shapeOptions.strokeWidth` option set to `0`. Then add a sample rectangle annotation and click the edit popup - **the input for editing the line width is not available**.
Suggested solution: change `option &&` to `defined(option) &&` here https://github.com/highcharts/highcharts/blob/061c33d702921510952a30a88ac0319a23912099/ts/Extensions/Annotations/NavigationBindings.ts#L827

- the `traverse` function takes only 4 parameters: https://github.com/highcharts/highcharts/blob/061c33d702921510952a30a88ac0319a23912099/ts/Extensions/Annotations/NavigationBindings.ts#L822
```js
function traverse(option, key, parentEditables, parent) {
```
so the fifth argument shouldn't be passed here: https://github.com/highcharts/highcharts/blob/061c33d702921510952a30a88ac0319a23912099/ts/Extensions/Annotations/NavigationBindings.ts#L912
```js
traverse(typeOption, typeKey, nestedEditables, visualOptions[key], true);
```

About making the `annotationsEditable` options public - we will think about it. I don't see any reason why we shouldn't do it.

**Comment by kamil-musialowski** [comment]
Check for `option` got fixed in #19494.
Traverse parameters amount also got fixed previously.

@pawelfus Could you please confirm, before closing it?
